pyproject.toml

```diff
@@ -0,0 +1,79 @@
+[tool.ruff]
+select = [
+  "A",      # flake8-builtins
+  "ARG",    # flake8-unused-arguments
+  "B",      # flake8-bugbear
+  "C4",     # flake8-comprehensions
+  "C90",    # McCabe cyclomatic complexity
+  "DTZ",    # flake8-datetimez
+  "E",      # pycodestyle
+  "EXE",    # flake8-executable
+  "F",      # Pyflakes
+  "G",      # flake8-logging-format
+  "I",      # isort
+  "ICN",    # flake8-import-conventions
+  "INP",    # flake8-no-pep420
+  "INT",    # flake8-gettext
+  "ISC",    # flake8-implicit-str-concat
+  "N",      # pep8-naming
+  "PGH",    # pygrep-hooks
+  "PIE",    # flake8-pie
+  "PL",     # Pylint
+  "PT",     # flake8-pytest-style
+  "PTH",    # flake8-use-pathlib
+  "PYI",    # flake8-pyi
+  "RET",    # flake8-return
+  "RSE",    # flake8-raise
+  "RUF",    # Ruff-specific rules
+  "S",      # flake8-bandit
+  "T10",    # flake8-debugger
+  "TCH",    # flake8-type-checking
+  "TID",    # flake8-tidy-imports
+  "UP",     # pyupgrade
+  "W",      # pycodestyle
+  "YTT",    # flake8-2020
+  # "ANN",  # flake8-annotations
+  # "BLE",  # flake8-blind-except
+  # "COM",  # flake8-commas
+  # "D",    # pydocstyle
+  # "DJ",   # flake8-django
+  # "EM",   # flake8-errmsg
+  # "ERA",  # eradicate
+  # "FBT",  # flake8-boolean-trap
+  # "NPY",  # NumPy-specific rules
+  # "PD",   # pandas-vet
+  # "Q",    # flake8-quotes
+  # "SIM",  # flake8-simplify
+  # "SLF",  # flake8-self
+  # "T20",  # flake8-print
+  # "TRY",  # tryceratops
+]
+ignore = [
+  "ARG002",
+  "B007",
+  "B904",
+  "E722",
+  "N811",
+  "N818",
+  "PIE790",
+  "PLR5501",
+  "PTH122",
+  "PTH123",
+  "S110",
+]
+line-length = 79
+target-version = "py37"
```

Comment:
actually we still support py36

Reply:
Unfortunately neither the CPython Team nor ruff supports end-of-life versions of Python.
* https://devguide.python.org/versions
* https://beta.ruff.rs/docs/settings/#target-version

Note that Python 3.7 will also go EOL in a few months.

Reply:
Ah, ok. Then never mind. I will just make sure that the tests run through locally before a release. I try to be a bit conservative with de-supporting Python versions in DBUtils since it's used in a few old-school projects running on LTS operating systems.

Reply:
But now that I think about it, let's move forward here. I have created a branch 3.0.x for supporting Python 3.6, so that we can modernize a bit more in the main branch. Maybe even adding type hints, using pytest and so on.